libs/core/src/lib/content-density/content-density.module.ts

```diff
@@ -0,0 +1,74 @@
+import { ModuleWithProviders, NgModule, Provider } from '@angular/core';
+import { CommonModule, DOCUMENT } from '@angular/common';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ContentDensityDirective } from './directives/content-density.directive';
+import { DEFAULT_CONTENT_DENSITY } from './tokens/default-content-density.token';
+import { ContentDensityMode, ContentDensityModuleConfig } from './content-density.types';
+import { ContentDensityStorage } from './classes/abstract-content-density-storage';
+import { MemoryContentDensityStorage } from './providers/memory-content-density-storage';
+import { ContentDensityControllerService } from './services/content-density-controller.service';
+import { LocalContentDensityStorage } from './providers/local-content-density-storage';
+import { CONTENT_DENSITY_STORAGE_KEY } from './tokens/content-density-storage-key.token';
+import { UrlContentDensityStorage } from './providers/url-content-density-storage';
+
+function generateContentDensityStorage(config: ContentDensityModuleConfig): Provider {
+    if (config.storage === 'localStorage') {
+        return {
+            provide: ContentDensityStorage,
+            useClass: LocalContentDensityStorage
+        };
+    }
+    if (config.storage === 'memory') {
+        return {
+            provide: ContentDensityStorage,
+            useClass: MemoryContentDensityStorage
+        };
+    }
+    if (config.storage === 'url') {
+        return {
+            provide: ContentDensityStorage,
+            useClass: UrlContentDensityStorage,
+            deps: [Router, ActivatedRoute, DEFAULT_CONTENT_DENSITY, CONTENT_DENSITY_STORAGE_KEY, DOCUMENT]
+        };
+    }
+    return [];
+}
+
+@NgModule({
+    imports: [CommonModule],
+    exports: [ContentDensityDirective],
+    declarations: [ContentDensityDirective]
+})
+export class ContentDensityModule {
```

Comment:
since module can be configured, can you please add an example for this?

Reply:
+1